XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs

```diff
@@ -171,9 +158,26 @@ protected void ShowWarning(string warningText)
         protected override void PageLoadedCore(PageLoadedDirection direction)
 		{
             ChosenItem = null;
-            restoreGridHomeServerSelection = (direction == PageLoadedDirection.Back);
+            restoreGridHomeServerSelection = direction == PageLoadedDirection.Back;
+            PopulateComboBox();
 		}
 
+        public override void PageCancelled()
+        {
+            UnregisterHandlers();
+            CancelFilters();
+            ClearComboBox();
+            ClearDataGridView();
+            ChosenItem = null;
+        }
+
+        protected override void PageLeaveCore(PageLoadedDirection direction, ref bool cancel)
+        {
+            UnregisterHandlers();

```

Comment:
This code is never reached by the Migrate VM wizard, because the `CrossPoolMigrateDestinationPage `class overrides `PageLeaveCore`, without calling the method in the base class.